### DISCUSSION TITLE
Enable touch simulation for sidebar webbrowser？

### DISCUSSION THREAD

**Discussion by benzBrake** [Floorp-Projects/Floorp#442] [Feature Request]
If I add a mobile-only web page to the sidebar（eg：https://tools-vue.zuoyebang.com/static/hy/tools-vue/calculator-3a903d48-hycache.html ）, all buttons on the web page cannot be clicked.

I don't know if it is difficult to add touch simulation. If it is not possible, I will find another webpage available on PC.

**Comment by black7375** [comment]
I think it's possible.
Firefox has this feature built in as Responsive design mode(`Ctrl+Shift+M`).
User agents for mobile are also available.

However, it will require some work to implement.

**Comment by Ryosuke-Asano** [comment]
@black7375 saying is true.

I found some code to relate about that. Implement is difficult. But I will try to add it

**Reply by black7375** [reply]
Implementation is likely to be tricky, as you'll need to load and use features from Devtools.

However, if you can, it's more likely that you'll be able to also get a tab sleep feature that stops the JavaScript timer like Edge does, rather than tab discard.
https://github.com/black7375/Firefox-UI-Fix/issues/262

It was originally a fairly old plan to [turn it into an extension and implement](https://github.com/black7375/Firefox-UI-Fix/discussions/502) it, but I ran out of time. :sob:

**Reply by Ryosuke-Asano** [reply]
I see.

 It seems possible to implement it, but it may take a while.

   I'm trying to implement embedding the browser window in the sidebar.

**Reply by Ryosuke-Asano** [reply]
A project that allows run content script (addon) on webpanel. After that, It can will be use 

https://github.com/Floorp-Projects/Floorp/commit/6eaf6b7b642f9afce8222b86bdbd9b1b29c8819f

**Comment by Ryosuke-Asano** [comment]
https://github.com/Floorp-Projects/Floorp/assets/73892113/d0c4b68d-0362-4e5c-a9f7-2c2423db028c

**Reply by black7375** [reply]
Looks cool feature!!
